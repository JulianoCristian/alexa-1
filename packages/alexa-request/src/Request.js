import { Request as Constants } from 'alexa-constants';

export default class Request {
  static session = (...args) => new Request().session(...args);
  static intent = (...args) => new Request().intent(...args);
  static launchRequest = (...args) => new Request().launchRequest(...args);
  static sessionEndedRequest = (...args) => new Request().sessionEndedRequest(...args);

  constructor(state = {}) {
    this.state = state;
  }

  session(session) {
    return new Request({
      ...this.state,
      session: {
        ...this.state.session,
        ...session
      }
    });
  }

  intent(name, slots = {}) {
    const slotData = Object.entries(slots).reduce((state, [name, value]) => ({ ...state, [name]: { name, value } }), {});
    return new Request({
      ...this.state,
      request: {
        type: Constants.Intent,
        intent: {
          name: name,
          ...(Object.keys(slotData).length ? { slots: slotData } : null)
        }
      }
    });
  }

  launchRequest() {
    return new Request({
      ...this.state,
      request: {
        type: Constants.Launch
      }
    });
  }

  sessionEndedRequest() {
    return new Request({
      ...this.state,
      request: {
        type: Constants.SessionEnded
      }
    });
  }

  build() {
    return { ...this.state };
  }
}
