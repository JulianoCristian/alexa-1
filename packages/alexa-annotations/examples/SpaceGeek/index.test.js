import Request from 'alexa-request';
import SpaceGeek from './index';

beforeEach(() => {
  jest.spyOn(Math, 'random').mockImplementation(() => 0.7123406182508916);
});

afterEach(() => {
  jest.resetAllMocks();
  jest.clearAllMocks();
});

test('LaunchRequest', () => {
  const event = Request.launchRequest().build();
  return expect(SpaceGeek(event)).resolves.toMatchSnapshot();
});

test('GetNewFactIntent', () => {
  const event = Request.intent('GetNewFactIntent').build();
  return expect(SpaceGeek(event)).resolves.toMatchSnapshot();
});
