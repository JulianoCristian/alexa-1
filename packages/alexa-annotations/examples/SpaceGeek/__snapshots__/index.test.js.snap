// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetNewFactIntent 1`] = `
Object {
  "response": Object {
    "card": Object {
      "content": "Here's your space fact: A total solar eclipse can happen once every 1 to 2 years. This makes them a rare event.",
      "title": "SpaceGeek",
      "type": "Simple",
    },
    "outputSpeech": Object {
      "text": "Here's your space fact: A total solar eclipse can happen once every 1 to 2 years. This makes them a rare event.",
      "type": "PlainText",
    },
    "shouldEndSession": true,
  },
  "version": "1.0",
}
`;

exports[`LaunchRequest 1`] = `
Object {
  "response": Object {
    "card": Object {
      "content": "Here's your space fact: A total solar eclipse can happen once every 1 to 2 years. This makes them a rare event.",
      "title": "SpaceGeek",
      "type": "Simple",
    },
    "outputSpeech": Object {
      "text": "Here's your space fact: A total solar eclipse can happen once every 1 to 2 years. This makes them a rare event.",
      "type": "PlainText",
    },
    "shouldEndSession": true,
  },
  "version": "1.0",
}
`;
